File: backend/app/schemas/reports.py
from datetime import datetime
from typing import Any, Literal
from uuid import UUID

from pydantic import BaseModel


class ReportHtmlResponse(BaseModel):
    calculation_run_id: UUID
    project_id: UUID
    scenario_id: UUID
    report_type: str
    title: str
    html: str
    context: dict[str, Any]


ExecutiveReportHtmlResponse = ReportHtmlResponse


class GenerateReportRequest(BaseModel):
    scenario_id: UUID
    calculation_run_id: UUID
    report_type: Literal["executive", "detailed"] = "executive"
    language: Literal["fr", "en"] = "en"
    branding_profile_id: UUID | None = None
    include_assumptions: bool = True
    include_regulatory_section: bool = False
    include_annexes: bool = True


class GeneratedReportResponse(BaseModel):
    id: UUID
    organization_id: UUID
    project_id: UUID
    scenario_id: UUID
    calculation_run_id: UUID
    branding_profile_id: UUID | None
    report_type: str
    status: str
    title: str
    file_name: str
    mime_type: str
    file_size_bytes: int
    generator_version: str
    created_at: datetime

File: frontend/types/reports.ts
export type ReportStatus = "pending" | "generated" | "failed";
export type ReportType = "executive" | "detailed";

export type GeneratedReportResponse = {
  id: string;
  organization_id: string;
  project_id: string;
  scenario_id: string;
  calculation_run_id: string;
  branding_profile_id: string | null;
  report_type: ReportType;
  status: ReportStatus | string;
  title: string;
  file_name: string;
  mime_type: string;
  file_size_bytes: number;
  generator_version: string;
  created_at: string;
};

export type CalculationResultLatestResponse = {
  calculation_run_id: string;
  project_id: string;
  scenario_id: string;
  status: string;
  engine_version: string;
  summary: {
    baseline_energy_kwh_year: number;
    scenario_energy_kwh_year: number;
    energy_savings_percent: number;
    baseline_bacs_class: string | null;
    scenario_bacs_class: string | null;
  };
  economic: {
    total_capex: number;
    annual_cost_savings: number;
    simple_payback_years: number;
    npv: number;
    irr: number;
  };
  messages: string[];
  warnings: string[];
  input_snapshot: Record<string, unknown>;
};
